src/media-segment-request.js

```diff
@@ -170,11 +170,23 @@ const handleKeyResponse = (segment, finishProcessingFn) => (error, request) => {
  *                                        this request
  */
 const handleInitSegmentResponse = (segment, finishProcessingFn) => (error, request) => {
+  const response = request.response;
   const errorObj = handleErrors(error, request);
 
   if (errorObj) {
     return finishProcessingFn(errorObj, segment);
   }
+
+  // stop processing if received empty content
+  if (response.byteLength === 0) {
```

Comment:
Is there any reason we wouldn't want to return this error for key requests? I think it would make sense to put this check at the end of `handleErrors` and just return the `errorObj` instead of making the check in both segment and init responses.